autotest/ogr/ogr_osm.py

```diff
@@ -979,9 +979,7 @@ def ogr_osm_18():
 
     ds = ogr.Open('data/two_points.pbf')
     lyr = ds.GetLayerByName('points')
-    count = 0
```

Comment:
this change should be revert. For performance reason GetFeatureCount() on a OSM layer (which is what is called with len(lyr)) returns -1.

Reply:
How is the following not semantically identical?

```
-    count = 0
-    for f in lyr:
-        count += 1
+    count = len(lyr)
```

Reply:
That is indeed delicious. :-)